2020/dec17/energysource/source3d/iteration.go

```diff
@@ -0,0 +1,49 @@
+package source3d
+
+// iterate once and return a new State
+func (s *State) iterate() *State {
+	next := New()
+	for z := s.min.z - 1; z <= s.max.z+1; z++ {
+		for y := s.min.y - 1; y <= s.max.y+1; y++ {
+			for x := s.min.x - 1; x <= s.max.x+1; x++ {
+				c := coord3D{x, y, z}
+				active := s.nextCellState(c)
+				if active {
+					next.Activate(x, y, z)
+				}
+			}
+		}
+	}
+	return next
+}
+
+// nextCellState returns the activation (or dis- thereof) of a cell: it returns the next state of this cell
+func (s *State) nextCellState(c coord3D) bool {
```

Comment:
technically could be
```suggestion
func (s State) nextCellState(c coord3D) bool {
```